packages/sdk/src/oracle/store/has.ts

```diff
@@ -0,0 +1,22 @@
+import Read from "./read";
+// we only ignore this specific error type explicitly emitted from reader class, meaning value is undefined
+import { ignoreExistenceError as ignoreError } from "../errors";
+import type { State } from "../types/state";
+
+// This class checks for existence for values you could potentially read. It mirrors the read interface but returns bools from functions.
+export default class Has {
```

Comment:
I think this has class is totally fine, but if the purpose of this class was to respond to my previous comments, you can feel free to remove. As long as we're catching a _specific type_ of error, I have no problem with how you were originally doing catches. Either way is fine with me, just wanted to mention.

Reply:
cool, yea i will keep the class as i think it will have utility outside of this. i was just avoiding having to write it and this just gave me the motivation to do it